core/src/main/scala/org/apache/spark/internal/io/HadoopMapReduceCommitProtocol.scala

```diff
@@ -167,8 +169,8 @@ class HadoopMapReduceCommitProtocol(
     committer.commitJob(jobContext)
 
     if (hasValidPath) {
-      val (allAbsPathFiles, allPartitionPaths) =
-        taskCommits.map(_.obj.asInstanceOf[(Map[String, String], Set[String])]).unzip
+      val (allAbsPathFiles, allPartitionPaths, successAttemptIDs) =
```

Comment:
Should we add new implicit function like '
```
implicit def asPair(x:(Map[String, String], Set[String], String)) = (x._1, (x._2, x._3))
```
before line 172, then we can `unzip` taskCommits as `(Map[String, String], (Set[String], String))` and reduce re-`zip` at line 188

Reply:
Sounds reasonable, I'll try implicit conversion